src/routes/users/sign-up-route.js

```diff
@@ -43,7 +55,23 @@ const signUpRoute = (request, response) => {
     request.on("end", function() {
       const user = JSON.parse(body);
       if (checkUser(user)) {
-        saveUser(user);
+        const fileUser = path.join(
+          __dirname,
+          "../../",
+          "db",
+          "/users",
+          `${user.username}.json`
+        );
+
+        fs.stat(fileUser, function(err, stat) {
+          if (err == null) {
+            console.log("Error! User already exists!");
```

Comment:
Not exactly. You need to send to user 409 error in such case. Also be careful with callback logic. It executes after the main route handler execution. Or better just move to promises logic